src/ropes/macros.rs

```diff
@@ -95,3 +95,88 @@ macro_rules! impl_rope {
         }
     }
 }
+
+
+/// Generate struct definition and implementation of RopeSlice
+///
+/// This is done through a macro because it relies on the Lnode type which is different
+/// depending on which type of Rope you are working with. This macro is used in both the
+/// rope & src_rope modules of this crate. In both places, the Lnode type from that
+/// module is passed in so we can generate the correct struct definition.
+macro_rules! generate_ropeslice_struct {
+    ($lnode: ty) => {
+        // A view over a portion of a Rope. Analagous to string slices (`str`);
+        pub struct RopeSlice<'rope> {
+            // All nodes which make up the slice, in order.
+            nodes: Vec<&'rope $lnode>,
+            // The offset of the start point in the first node.
+            start: usize,
+            // The length of text in the last node.
+            len: usize,
+            // The index of the current byte - only used for iterating
+            cur_byte: usize,
+            // The index of the current node - only used for iterating
+            cur_node: usize,
+        }
+
+        impl<'rope> RopeSlice<'rope> {
+            fn empty<'r>() -> RopeSlice<'r> {
+                RopeSlice {
+                    nodes: vec![],
+                    start: 0,
+                    len: 0,
+                    cur_node: 0,
+                    cur_byte: 0,
+                }
+            }
+        }
+
+        impl<'rope> Iterator for RopeSlice<'rope> {
+            type Item = u8;
```

Comment:
I'm a bit surprised you want to iterate over the bytes and not the characters. Why do you want the bytes?